controllers/cluster_controller.go

```diff
@@ -262,9 +270,135 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 		}
 	}
 
+	if cluster.IsPackagesEnabled() {
+		if err := r.postReconcilePackagesForWorkloadCluster(ctx, log, cluster); err != nil {
+			return controller.Result{}, err
+		}
+	}
+
 	return controller.Result{}, nil
 }
 
+// PackagesClient handles curated packages operations from within the cluster
+// controller.
+type PackagesClient interface {
```

Comment:
Any reason this is defined here? It might make more sense to define it somewhere up there or where it is getting used.

Reply:
Just let me know where you'd like it to be, and I'll move it there. It's used in several places, and "up there" is rather vague, but if you can give me a line number in this file, I'll try to get it as close as I can.